app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CommonController.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import play.api.Logger
+import play.api.mvc.Result
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import uk.gov.hmrc.tariffclassificationfrontend.model.{ErrorCode, JsErrorResponse}
+
+trait CommonController extends FrontendController {
+
+  private[controllers] def recovery: PartialFunction[Throwable, Result] = {
+    case e: Throwable =>
+      Logger.error(s"Error occurred: ${e.getMessage}", e)
+      handleException(e)
+  }
+
+  private[controllers] def handleException(e: Throwable) = {
```

Comment:
Is there not a better way to do this with Play middleware? e.g. https://www.playframework.com/documentation/2.6.x/ScalaErrorHandling